server/grpc_service.go

```diff
@@ -664,6 +664,22 @@ func (s *GrpcServer) StoreHeartbeat(ctx context.Context, request *pdpb.StoreHear
 		ClusterVersion:    rc.GetClusterVersion(),
 	}
 	rc.GetUnsafeRecoveryController().HandleStoreHeartbeat(request, resp)
+
+	// If this cluster has slow stores, we should awaken hibernated regions in other stores.
+	if needAwaken, slowStoreIDs := rc.NeedAwakenAllRegionsInStore(storeID); needAwaken {
+		log.Info("forcely awaken hibernated regions", zap.Uint64("store-id", storeID))
+		err := rc.UpdateAwakenStoreTime(storeID, time.Now())
+		if err != nil {
+			log.Warn("failed to awaken hibernated regions in store", zap.Uint64("store-id", storeID))
+		} else {
+			if slowStoreIDs == nil {
```

Comment:
Should this branch be blocked by `NeedAwakenAllRegionsInStore`?

Reply:
Yes, this judgement is redundant.